### PR TITLE
Fix assessment reporting error

### PR DESCRIPTION
**Pull request by breville**
It turns out that we sometimes move multi-page assessments between scripts.  The dashboard API to roll up assessment results was hitting an error when it encountered a student result in such a level.  Now, we reuse an existing retrieval of the UserLevel that doesn't check which script the level was in when submitted, meaning that we report back the student's results for that assessment regardless of which script contained it.

### CODE REVIEW

File: dashboard/app/controllers/api_controller.rb [hunk 1]
```diff
@@ -253,6 +253,8 @@ def section_assessments
 
         user_level = student.user_level_for(script_level, level_group)
 
+        next unless user_level
```
Comment (ashercodeorg) [inline dashboard/app/controllers/api_controller.rb]: Can line 304 (alternately line 316) now be changed to user_level.submitted?

Comment (ashercodeorg) [reply dashboard/app/controllers/api_controller.rb]: Is there any reason why lines 241 and 244 can't be filtered via line 235?

**Comment by ashercodeorg** [comment]
Mind assigning someone else in addition to me? At the moment, I'm having enough difficulty understanding (a) the context, (b) why or how `user_level` might be null, and (c) whether we are sweeping something under the rug by skipping them.

**Comment by breville** [comment]
Thanks for pushing back on this Asher.  After we got a concrete repro, I investigated and discovered that the assessment had been moved between scripts, leading to a better fix in which we no longer care which script contained the LevelGroup when the user responded to it.

File: dashboard/app/controllers/api_controller.rb [hunk 2]
```diff
@@ -243,16 +243,22 @@ def section_assessments
         # Don't allow somebody to peek inside an anonymous survey using this API.
         next if script_level.anonymous?
 
+        # Get the UserLevel for the last attempt.  This approach does not check
+        # for the script and so it'll find the student's attempt at this level for
+        # any script in which they have encountered that level.
         last_attempt = student.last_attempt_for_any(script_level.levels)
+
+        # Get the Level itself.
```
Comment (ashercodeorg) [inline dashboard/app/controllers/api_controller.rb]: This comment (which I believe to be correct) is at odds with the variable name.

EDIT: Never mind on the "believe to be correct" part.

File: dashboard/app/controllers/api_controller.rb [hunk 3]
```diff
@@ -243,16 +243,22 @@ def section_assessments
...
         level_group = last_attempt.try(:level) || script_level.oldest_active_level
+
+        # Get the response which will be stringified JSON.
         response = last_attempt.try(:level_source).try(:data)
 
         next unless response
 
+        # Parse the response string into an object.
```
Comment (ashercodeorg) [inline dashboard/app/controllers/api_controller.rb:259]: Though I appreciate the comment (not suggesting you remove it), the code is 100% as clear as the comment.

**Comment by balderdash** [comment]
LGTM

**Pull request merged**